wyvern/components/models/base_ranking_model.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+import asyncio
+from typing import List, Union
+
+from wyvern.components.models.model_component import (
+    MODEL_INPUT,
+    MODEL_OUTPUT,
+    ModelComponent,
+    SingleEntityInput,
+)
+
+
+class BaseRankingModel(ModelComponent[MODEL_INPUT, MODEL_OUTPUT]):
+    async def single_inference(
+        self,
+        input: SingleEntityInput,
+    ) -> Union[float, str, List[float]]:
+        """
+        Single inference for ranking model
+        """
+        raise NotImplementedError
+
+    async def batch_inference(
+        self,
+        input: MODEL_INPUT,
+    ) -> MODEL_OUTPUT:
+        """
+        Batch inference for ranking model
+        """
+        model_score_batches = await asyncio.gather(
+            *[
+                self.single_inference(
```

Comment:
This is going to cause performance issues for sure, either via network fanout or by capturing a lot of overhead for single inference calls.

From a use-case perspective:
1. Things like Fraud will always be doing a single inference, so it's not a big deal there
2. Things like Ranking wil lalways be doing batch inference, so this default is going to be a big problem

What do you think about inverting this design? `single_inference` calls `batch_inference` with a size of 1?

Reply:
I completely removed BaseRankingModel.

https://github.com/Wyvern-AI/wyvern/pull/25/files#diff-2d2d1b0993399a36b7a3167d82e34a8ca801ef45239db1137f93297ce46a66b6R149-R152

we should always prioritize batch_inference.
for models like fraud, they could just define one `async def single_inference` and the model is done.

for ranking, they should always redefine the `async def batch_inference`

now the `async def inference` interface truncate the big entity list into smaller batch size so that the batch_inference actually just take care of one batch data.